extensions/redis-client/runtime/src/main/java/io/quarkus/redis/client/runtime/RedisClientUtil.java

```diff
@@ -60,4 +62,59 @@ public static boolean isDefault(String clientName) {
     public static RedisConfiguration getConfiguration(RedisConfig config, String name) {
         return isDefault(name) ? config.defaultClient : config.additionalRedisClients.get(name);
     }
+
+    /**
+     * This method was only added to support minimal backward compatibility.
+     * <p>
+     * It should be removed in the 1.10 release.
+     * Follows up https://github.com/quarkusio/quarkus/pull/11908#issuecomment-694794724
+     */
+    private static String buildConnectionString(RedisConfiguration config, URI host, String clientName) {
+        final String address = host.toString();
+        if (address.contains("://")) {
+            return address;
```

Comment:
I /think/ I would throw an error here if any of the deprecated Optional bits are defined.

Reply:
If we throw an error we won't be backward compatible anymore.

Reply:
Did we support having `://` in the address before?

Reply:
If we did, yes, what I suggested is stupid :). If we didn't, it makes sense :).

Reply:
No we did not.  The scheme got introduced only with the new way of configuring the Redis connection string. 

Meaning that if the host does not contain it, the it's the old way of doing things and throwing an error won't be backward compatible anymore. Hence me the need to re-construct the connection string as it was previous done. 

Maybe I am missing something.

Reply:
I added my comment at exactly the place I wanted us to throw the error.

If address contains `://` then it's the new way of doing things so, **in this case** if any of these fields is defined we could throw an error? Or am I missing something?

Reply:
(Note that I might be missing something, I'm exhausted :))

Reply:
> I added my comment at exactly the place I wanted us to throw the error.
> 
> If address contains `://` then it's the new way of doing things so, **in this case** if any of these fields is defined we could throw an error? Or am I missing something?

I get your comment now. 

Sorry it took me long to realize it since the comment appear just after the `return address;` statement  and I read your comment all wrong - (that it concerned the old way).

Yes, you are right we could throw.  I think the old way made it impossible to define the host with the `scheme`  i.e `://` part. So if they enter that line, it is by following the *new way* of doing things and chances are low that they'll mix it with the deprecated configs since they'll be hidden. If it happens to be the case,  I think we can be lenient here and just start the application with the right configuration and ignore the old ones. They'll get a nice warning message if they keep on using the old property once we remove them post `1.9.x` release. 

Hope it makes sense.